#  SEMgraph library
#  Copyright (C) 2019-2021 Fernando Palluzzi; Mario Grassi
#  e-mail: <fernando.palluzzi@gmail.com>
#  University of Pavia, Department of Brain and Behavioral Sciences
#  Via Bassi 21, 27100 Pavia, Italy

#  SEMgraph is free software: you can redistribute it and/or modify
#  it under the terms of the GNU General Public License as published by
#  the Free Software Foundation, either version 3 of the License, or
#  (at your option) any later version.

#  SEMgraph is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.

#  You should have received a copy of the GNU General Public License
#  along with this program.  If not, see <https://www.gnu.org/licenses/>.

# -------------------------------------------------------------------- #

#' @title KEGG interactome
#'
#' @description Interactome generated by merging KEGG pathways 
#' extracted using the \code{ROntoTools} R package (update: November, 2021).
#' @name kegg
#' @usage kegg
#' @docType data
#' @format
#' "kegg" is an igraph network object of 5007 nodes and 44755 edges 
#' (41122 directed and 3164/2 = 1582 bidirected) corresponding to the
#' union of 225 KEGG pathways.
#' @source \url{https://www.genome.jp/kegg/}
#' @references
#' 
#' Kanehisa M, Goto S (1999). KEGG: kyoto encyclopedia of genes and genomes. 
#' Nucleic Acid Research 28(1): 27-30. 
#' <https://doi.org/10.1093/nar/27.1.29>
#'
#' Calin Voichita, Sahar Ansari and Sorin Draghici (2021).
#' ROntoTools: R Onto-Tools suite. R package version 2.20.0.
#' 
#' @examples
#' 
#' \donttest{
#' library(SEMgraph)
#' 
#' #KEGG graph
#' properties(kegg)
#' 
#' }
#'
NULL

#' @title Reactome interactome
#'
#' @description Interactome generated by merging Reactome pathways 
#' extracted using the \code{graphite} R package (update: April, 2020).
#' @name reactome
#' @usage reactome
#' @docType data
#' @format
#' "reactome" is an igraph network object of 9762 nodes and 416128 edges 
#' (109268 directed and 306860/2 = 153430 bidirected) corresponding to the union
#' of 1641 Reactome pathways.
#' @source \url{https://reactome.org/}
#' @references
#' 
#' Jassal B, Matthews L, Viteri G, Gong C, Lorente P, Fabregat A, 
#' Sidiropoulos K, Cook J, Gillespie M, Haw R, Loney F, May B, Milacic M, 
#' Rothfels K, Sevilla C, Shamovsky V, Shorser S, Varusai T, Weiser J, 
#' Wu G, Stein L, Hermjakob H, D'Eustachio P. The reactome pathway knowledgebase. 
#' Nucleic Acids Res. 2020 Jan 8;48(D1):D498-D503.
#' <https://doi:10.1093/nar/gkz1031>.
#' 
#' Sales G, Calura E, Cavalieri D, Romualdi C (2012). graphite - a Bioconductor
#' package to convert pathway topology to gene network. BMC Bioinformatics 2012, 13:20.
#' <https://doi:10.1186/1471-2105-13-20>. 
#' 
#' @examples
#' 
#' \donttest{
#' library(SEMgraph)
#'
#' #Reactome graph
#' properties(reactome)
#' 
#' }
#'
NULL

#' @title DO interactome
#'
#' @description DO interactome
#' @name ig.DO
#' @usage ig.DO
#' @docType data
#' @format
#' "ig.DO" is an igraph network object of 6576 nodes and 6923 directed edges, 
#' corresponding to the Disease Ontology terms (update: Feb, 2020).
#' It has attributes to vertices/nodes and edges:
#' \enumerate{
#'  \item "name" (v/c), i.e., "Term ID";
#'  \item "term_id" (v/c), i.e., "Term ID";
#'  \item "term_name" (v/c), i.e,. "Term Name";
#'  \item "term_distance" (v/c), i.e., Term Distance: the distance to the root; always 0 for the root itself;
#'  \item "relation" (e/c), either 'is_a' or 'part_of'.
#' }
#' @source \url{https://disease-ontology.org/downloads/}
#' @references
#' Schriml et al. Disease Ontology: a backbone for disease semantic integration. Nucleic Acids Res 2012, 40:D940-946.
#' <https://doi.org/10.1093/nar/gkr972>
#'
#' @examples
#' 
#' \donttest{
#' library(SEMgraph)
#' 
#' # DO graph
#' properties(ig.DO)
#'
#' }
#'
NULL

#' @title GOBP interactome
#'
#' @description GOBP interactome
#' @name ig.GOBP
#' @usage ig.GOBP
#' @docType data
#' @format
#' "ig.GOBP" is an igraph network object of 27685 nodes and 58979 directed edges,
#' corresponding to the Gene Ontology Biological Process terms (update: Feb, 2020).
#' It has attributes to vertices/nodes and edges:
#' \enumerate{
#'  \item "name" (v/c), i.e., "Term ID";
#'  \item "term_id" (v/c), i.e., "Term ID";
#'  \item "term_name" (v/c), i.e,. "Term Name";
#'  \item "term_distance" (v/c), i.e., Term Distance: the distance to the root; always 0 for the root itself;
#'  \item "relation" (e/c), either 'is_a' or 'part_of'.
#' }
#' @source \url{https://geneontology.org/docs/download-ontology/}
#' @references
#' Ashburner et al. Gene ontology: tool for the unification of biology. Nat Genet 2000, 25:25-29.
#' <https://doi.org/10.1038/75556>
#'
#' @examples
#' 
#' \donttest{
#' library(SEMgraph)
#' 
#' # GOBP graph
#' properties(ig.GOBP)
#'
#' }
#'
NULL

#' @title GOCC interactome
#'
#' @description GOCC interactome
#' @name ig.GOCC
#' @usage ig.GOCC
#' @docType data
#' @format
#' "ig.GOCC" is an igraph network object of 3793 nodes and 7116 directed edges,
#' corresponding to the Gene Ontology Cellular Component terms (update: Feb, 2020).
#' It has attributes to vertices/nodes and edges:
#' \enumerate{
#'  \item "name" (v/c), i.e., "Term ID";
#'  \item "term_id" (v/c), i.e., "Term ID";
#'  \item "term_name" (v/c), i.e,. "Term Name";
#'  \item "term_distance" (v/c), i.e., Term Distance: the distance to the root; always 0 for the root itself;
#'  \item "relation" (e/c), either 'is_a' or 'part_of'.
#' }
#' @source \url{https://geneontology.org/docs/download-ontology/}
#' @references
#' Ashburner et al. Gene ontology: tool for the unification of biology. Nat Genet 2000, 25:25-29.
#' <https://doi.org/10.1038/75556>
#'
#' @examples
#' 
#' \donttest{
#' library(SEMgraph)
#' 
#' # GOCC graph
#' properties(ig.GOCC)
#' 
#' }
#'
NULL

#' @title GOMF interactome
#'
#' @description GOMF interactome
#' @name ig.GOMF
#' @usage ig.GOMF
#' @docType data
#' @format
#' "ig.GOMF" is an igraph network object of 9933 nodes and 12114 directed edges, 
#' corresponding to the Gene Ontology Molecular Function terms (update: Feb, 2020).
#' It has attributes to vertices/nodes and edges:
#' \enumerate{
#'  \item "name" (v/c), i.e., "Term ID";
#'  \item "term_id" (v/c), i.e., "Term ID";
#'  \item "term_name" (v/c), i.e,. "Term Name";
#'  \item "term_distance" (v/c), i.e., Term Distance: the distance to the root; always 0 for the root itself;
#'  \item "relation" (e/c), either 'is_a' or 'part_of'.
#' }
#' @source \url{https://geneontology.org/docs/download-ontology/}
#' @references
#' Ashburner et al. Gene ontology: tool for the unification of biology. Nat Genet 2000, 25:25-29.
#' <https://doi.org/10.1038/75556>
#'
#' @examples
#' 
#' \donttest{
#' library(SEMgraph)
#' 
#' # GOMF graph
#' properties(ig.GOMF)
#' 
#' }
#'
NULL

#' @title STRING interactome
#'
#' @description STRING interactome
#' @name ig.STRING
#' @usage ig.STRING
#' @docType data
#' @format
#' "ig.STRING" is an igraph network object of 10829 nodes and 213445 edges 
#' (undirected), corresponding to the STRING interactome (version 10.5).
#' Only edges with heigh confidence (combined score>=900) are retained.
#' @source \url{https://string-db.org/}
#' @references
#' Szklarczyk D, Gable AL, Lyon D, Junge A, Wyder S, Huerta-Cepas J, 
#' Simonovic M, Doncheva NT, Morris JH, Bork P, Jensen LJ, von Mering C. 
#' STRING v11: protein-protein association networks with increased 
#' coverage, supporting functional discovery in genome-wide experimental 
#' datasets. Nucleic Acids Res. 2019, 47: D607-613. 
#' <https://doi.org/10.1093/nar/gky1131>
#'
#' @examples
#' 
#' \donttest{
#' library(SEMgraph)
#' 
#' # STRING graph
#' properties(ig.STRING)
#' 
#' }
#'
NULL
